driver/src/test/scala/za/co/absa/hyperdrive/driver/drivers/KafkaToKafkaDeduplicationAfterRetryDockerTest.scala

```diff
@@ -201,19 +203,47 @@ class KafkaToKafkaDeduplicationAfterRetryDockerTest extends FlatSpec with Matche
 
     exceptionWasThrown shouldBe true
     fs.exists(new Path(s"$checkpointDir/$sourceTopic")) shouldBe true
+  }
+  // scalastyle:on method.length
 
-    val consumer = createConsumer(kafkaSchemaRegistryWrapper)
-    val records = getAllMessages(consumer, destinationTopic)
-
-    val valueFieldNames = records.head.value().getSchema.getFields.asScala.map(_.name())
-    valueFieldNames should contain theSameElementsAs List("record_id", "value_field", "hyperdrive_id")
-    val actualRecordIds = records.map(_.value().get("record_id"))
-    val expectedRecordIds = 0 until recordsV1.size + recordsV2.size
-    actualRecordIds should contain theSameElementsAs expectedRecordIds
+  before {
+    baseDir = TempDirectory("hyperdriveE2eTest").deleteOnExit()
   }
 
   after {
     SchemaManagerFactory.resetSRClientInstance()
+    baseDir.delete()
+  }
+
+  private def schemaV1String(name: String) =
+    raw"""{"type": "record", "name": "$name", "fields": [
+         |{"type": "int", "name": "record_id"},
+         |{"type": "string", "name": "value_field", "nullable": false}
+         |]}""".stripMargin
+
+  private def schemaV2String(name: String) =
+    raw"""{"type": "record", "name": "$name", "fields": [
+         |{"type": "int", "name": "record_id"},
+         |{"type": ["null", "string"], "name": "value_field", "nullable": true}
+         |]}""".stripMargin
+
+  private def sendData(producer: KafkaProducer[GenericRecord, GenericRecord], records: Seq[GenericRecord], topic: String, partitions: Int): Unit = {
+    records.zipWithIndex.foreach {
+      case (record, i) =>
+        val partition = i % partitions
+        val producerRecord = new ProducerRecord[GenericRecord, GenericRecord](topic, partition, null, record)
+        producer.send(producerRecord)
+    }
+  }
+
+  private def createTopic(kafkaSchemaRegistryWrapper: KafkaSchemaRegistryWrapper, topicName: String, partitions: Int): Unit = {
+    val config = new Properties()
+    config.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaSchemaRegistryWrapper.kafka.getBootstrapServers)
+    val localKafkaAdmin = AdminClient.create(config)
+    val replication = 1.toShort
+    val topic = new NewTopic(topicName, partitions, replication)
+    val topicCreationFut = localKafkaAdmin.createTopics(util.Arrays.asList(topic)).all()
+    while(!topicCreationFut.isDone) {}
```

Comment:
What about Await instead of while ?

Reply:
Absolutely. Forgot to change this here, there is `.get()` on the `KafkaFuture` which is blocking, so need for `.isDone` in the first place.